### PR TITLE
Incident command article

### PR DESCRIPTION
**Pull request by PJHGB**
Please have a review gents (and correct all my typo's)

### CODE REVIEW

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+layout:     post
+title:      Lessons Learned for Incident Commanders
+date:       2017-01-30 09:00
+summary:    Incident command is a reasonably new area of focus for SBG. In a nutshell we have a nominated technical person known as the Incident Commander (IC) who gives direction in order to resolve an incident and restore service as quickly as possible.<br><br>This blog post contains some of the insights and 'lessons learned' by our teams from their experiences in live incidents and exercises (known internally as fire drills) as they work to improve their skills and reduce our MTTR.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Might recommend defining MTTR or using `<abbr title="Mean time to resolution">MTRR</abbr>`

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+category:   Incident Response
+tags:       Incident management, Incident command
+author:     patrick_holmes
+image:      peace_war_inccmd.jpg
+---
+
+Incident command is a reasonably new area of focus for SBG. In a nutshell we have a nominated technical person known as the Incident Commander (IC) who gives direction in order to resolve an incident and restore service as quickly as possible.
+
+This blog post contains some of the insights and 'lessons learned' by our teams from their experiences in live incidents and exercises (known internally as fire drills) as they work to improve their skills and reduce our MTTR. The methodology for this is largely forked from O'Reilly's highly recommended [Incident Management for Operations](http://shop.oreilly.com/product/0636920036159.do) book, influences from SAR/British Army doctrine, with some additions based on some of our experiences across the company.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: SAR?

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+***
+
+#### DON'T PANIC!
+It’s just a computer system and a technical problem that needs solving, the only difference between this and a JIRA story is the manner of approach and doing it with urgency. Take a ‘Condor moment’ then crack on with systematically triaging your way through the issues and bringing the impacted services back online.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Backticks or apostrophes?

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+#### The first responder is the first incident commander
+The first person on the scene is by default in control of the incident, until until they get more help or can safely hand it off to someone else. In some cases this may mean they are best focusing on raising the alarm rather than immediately fighting the fire, but in all cases they need to focus effort to gain control of the situation.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "until until"

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "focusing" and "focus" in the same sentence is a bit repetitive.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Seize the initiative
+Make the mental switch from ‘peace’ to ‘war’ and do not allow the situation and events that are occurring to dictate the pace and direction of the incident. A calm and confident incident commander takes control of the situation and issues clear and explicit directives to his or her available technical resource in order to triage the incident and drive it towards resolution as quickly and safely as possible with the minimum amount of collateral damage. This is often the difference between a ten minute incident and a two hour incident.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "his or her" - in previous posts we have preferred the gender-neutral form of "their"

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "ten-minute" and "two-hour" - these are compound adjectives in this instance

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+#### IC’s do not fix problems
+As coordinators of resource the IC should not be at the terminal issuing commands. It is the role of the IC to set the strategic tone for the incident and leave the individual tactics to the subject matter experts at their disposal. In many cases, the IC has the team debate the pro's and con's of a particular solution – but the IC makes the final call based on the best available information to them from the SME's at the time.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "pros and cons" - no apostrophe necessary.

SME is undefined - again consider an `abbr` or expanding it followed by the abbreviation in parens.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Have a single point of contact for each tribe/function
+Best case scenario, the single point of contact in a team will be that teams subject matter expert who has also trained as an IC, they will understand the same processes the IC is going through and have information on hand before the IC needs it. Often this is the difference between the IC being able to maintain control of an incident or it descending into floods of Slack messages from 10 people in each team which will rapidly overwhelm them, and inevitably slows down the resolution of the incident. Often the fastest way to get the relevant single point of contact for each team is request that they dial into a conference call, people tend to be more reluctant to shout over the IC or others on a call
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "Best case scenario" - consider expanding this abbreviation to "In the best-case scenario" - just a personal preference on phrasing for long-form writing there.

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "that team's" needs an apostrophe - also expanded to "subject matter expert" here, but SME above. The other way round would make more sense.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Keep a log
+Whether on paper or on a computer. Developing and rehearsing your distributed ways of working and communicating using a tool like Slack will help with this if you keep the channel updated on what is going on. If you have taken your incident response to a conference call summarise it for the people not on the call regularly.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Dangling modifier! Do you summarise regularly or are people not on the call regularly? Consider rephrasing.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Clear, concise communication
+While in an emergency incident courtesy can often take a back seat to resolving the issue, this is a matter of priorities and will differ from normal BAU situations. Instructions, objectives and feedback should be delivered in a clear and specific manner, small talk and 'niceties' are kept to the minimum.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Could perhaps use a comma before "courtesy"

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: BAU

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: niceties is a regular word - it doesn't need quotes.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+- Assigning tasks to a specific person (by name) or function,
+- Stating clearly the objective,
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Yoda-wording. "Stating the objective clearly".

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+- Expecting the receiver to acknowledge and repeat back a summary of the assignment, and
+- Giving a specific time to complete the task.
+
+> IC: “Network ops, International tribe are seeing are seeing packet loss on their office VPN to the DC. Can you drop off the call, take a look and get back to me in at 0915?”
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "in at"

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+> **A:** “IC: Network ops have looked at it and are 90% certain that it is a problem with a runaway process on the firewall. They are currently working to fail over the cluster to resolve this within the next 10 minutes.”
+
+> **N:** “IC: I want to keep everyone on the conf call for now until we are certain of the fix. While we are waiting, lets discuss our next most likely root cause candidate and what our plan B will be.”
+
+#### On that subject, have a plan B!
+When your team have come up with plan A and are focussing on that you need to be thinking about plan B and C, getting resource lined up to support them and switching assets to support a secondary or tertiary plan when appropriate in a timely manner. Do not be afraid to adapt or change your plans where it is justified to do so. Your main focus should always be on restoring service as quickly as possible and it can sometimes be a tough call whether to ask a team to 'switch fire' to another plan. Where possible have multiple teams working on different plans rather than having a single team context switching to come up with multiple plans.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "focusing" used before and in previous posts on this blog. Drop the second "S" from this instance.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Don’t forget the customer!
+Our Service Lifecycle Managers are our IC's interface into our service operations and the customer facing teams, if you don't have this luxury then you can nominate someone to regularly 'check in' with the business and customer stakeholders and make sure you know whether your customers are contacting you.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: Not sure "check in" requires quotes.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+
+Don't forget you can use proactively use both technical and non-technical resources such as maintenance pages or/outage messaging systems and issue either internal or external communications which may help a potential problem from getting worse. Good external communications help your customers understand what is going on without them contacting your customer service teams en masse to find out why their account isn't working!
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "use proactively use"

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "or/outage"

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "en masse" - generally accepted semantic HTML usage for this kind of thing these days is `<i lang="fr">en masse</i>`. Happy to be corrected by any i18n / a11y experts though.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+#### If you need more resources or support, ask!
+An IC is responsible for driving the resolution of an incident, as such they can ask for whatever resources they require in order to achieve this objective of restoring service. This is not governed by normal ‘peacetime’ structure rules and the rest of the company should support this by whatever means are available. However it worth noting: While it is possible for IC’s to instruct senior directors in an incident they should bear in mind that this should be done respectfully and that they may be asked to explain their reasoning and thought process in a post-mortem at a later date. Bear in mind tomorrow is back to BAU!
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "While" doesn't need to be capitalised.

File: _posts/2017-01-30-inc-cmd-lessonslearned.md [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Practise, Practise, Practise
+Practise is the key to getting this right, it's always a fine balance and there is rarely such a thing as a straightforward problem. Your first attempt at incident command will (to borrow a metaphor from R4) most likely have all the co-ordination, control and precision of a pigeon trapped in a bottle bank. Your second one will be better, and every time subsequently will be incrementally better as you apply the lessons you have learned. People who have been doing this for 10-15 years will always walk away from an incident with for next time. Always factor lessons learned into a post-incident review (see [Etsy's debriefing facilitation guide](https://codeascraft.com/2016/11/17/debriefing-facilitation-guide/)), share them with the rest of your teams, and let everyone learn from your experience.
```
Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "Practice" the noun, not "practise", the verb.

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: R4?

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "will always walk away from an incident with for next time" ??

Comment (mitchellrj) [inline _posts/2017-01-30-inc-cmd-lessonslearned.md]: "lessons-learned"

**Comment by PJHGB** [comment]
Thanks @mitchellrj, corrected all of those. Interested in opinion on 'lessons learned' vs 'lessons-learned', aware I seem to use a mix in this now?

File: _posts/2017-02-02-inc-cmd-lessonslearned.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Clear, concise communication
+While in an emergency incident, courtesy can often take a back seat to resolving the issue and restoring service, this is a matter of priorities and will differ from normal <abbr title="Business as Usual">BAU</abbr> situations. Instructions, objectives and feedback should be delivered in a clear and specific manner, small talk and niceties are kept to a minimum.
+
+- Assigning tasks to a specific person (by name) or function,
+- Stating the objective clearly,
+- Expecting the receiver to acknowledge and repeat back a summary of the assignment, and
+- Giving a specific time to complete the task.
+
+> IC: “Network ops, International tribe are seeing are seeing packet loss on their office VPN to the DC. Can you drop off the call, take a look and get back to me at 0915?”
```
Comment (mattkirwan) [inline _posts/2017-02-02-inc-cmd-lessonslearned.md]: Multiple "are seeing"

Comment (PJHGB) [inline _posts/2017-02-02-inc-cmd-lessonslearned.md]: good spot, fixed, thanks!

**Pull request merged**